mev-relay-rs/src/relay.rs

```diff
@@ -568,21 +564,9 @@ impl BlindedBlockProvider for Relay {
                     warn!(%err, %auction_request, %block_root, "block failed beacon node validation");
                     Err(RelayError::InvalidSignedBlindedBeaconBlock.into())
                 } else {
-                    let local_payload = &auction_context.execution_payload;
-                    let block_hash = local_payload.block_hash();
+                    let block_hash = auction_context.execution_payload().block_hash();
                     info!(%auction_request, %block_root, %block_hash, "returning local payload");
-                    let auction_contents = match local_payload.version() {
-                        Fork::Bellatrix => AuctionContents::Bellatrix(local_payload.clone()),
-                        Fork::Capella => AuctionContents::Capella(local_payload.clone()),
-                        Fork::Deneb => {
-                            AuctionContents::Deneb(auction_contents::deneb::AuctionContents {
-                                execution_payload: local_payload.clone(),
-                                // TODO: support blobs
-                                blobs_bundle: Default::default(),
-                            })
-                        }
-                        _ => unreachable!("fork not reachable from type"),
-                    };
+                    let auction_contents = auction_context.to_auction_contents();
```

Comment:
bikeshedding but a bit more idiomatic imo is to call this method `into_contents`